freeipa/src/main/resources/freeipa-salt/salt/modifyproxy/init.sls

```diff
@@ -0,0 +1,83 @@
+{% if salt['pillar.get']('proxy') != None %}
+
+{%- from 'modifyproxy/settings.sls' import proxy with context %}
+
+{% if proxy.tunnel == 'CCM' %}
+
+{% if proxy.user %}
+
+ccm_ssh_proxy_auth:
+  file.managed:
+    - name: /root/.ssh/proxy_auth
+    - create: True
+    - mode: 600
+    - contents:
+      - "{{ proxy.user }}:{{ proxy.password }}"
+
+{% endif %} # proxy_user
+
+ccm_setup_ssh_proxy:
+  file.replace:
+    - name: /root/.ssh/config
+    - pattern: "^ProxyCommand"
+    - repl: "{{ proxy.proxy_command }}"
+    - append_if_not_found: True
+
+{% elif proxy.tunnel == 'CCMV2' or proxy.tunnel == 'CCMV2_JUMPGATE' %}
+
+ccmv2_setup_jumpgate_proxy:
+  file.replace:
+    - name: /etc/jumpgate/config.toml
+    - pattern: "^http_proxy.*"
+    - repl: 'http_proxy = "{{ proxy.proxy_url }}"'
+    - append_if_not_found: True
+
+{% endif %} # connectivity_mode
+
+proxy_env:
+  file.managed:
+    - name: /etc/cdp/proxy.env
+    - mode: 640
+    - makedirs: True
+    - contents:
+      - "https_proxy={{ proxy.proxy_url }}"
+{% if proxy.no_proxy_hosts %}
+      - "no_proxy={{ proxy.no_proxy_hosts }}"
+{% endif %}
+
+{% else %}
+# The proxy is removed
+
+ccm_remove_ssh_proxy:
+  file.replace:
+    - name: /root/.ssh/config
+    - pattern: "^ProxyCommand"
+    - repl: ""
+    - ignore_if_missing: True
+
+remove_proxy_env:
+  file.absent:
+    - name: /etc/cdp/proxy.env
+
+ccmv2_remove_jumpgate_proxy:
+  file.replace:
+    - name: /etc/jumpgate/config.toml
+    - pattern: "^http_proxy"
+    - repl: 'http_proxy = ""'
+
+{% endif %}
+
+restart_ccm_gateway_tunnel_service:
+  cmd.run:
+    - name: systemctl restart ccm-tunnel@GATEWAY
+    - onlyif: systemctl list-unit-files | grep "^ccm-tunnel@GATEWAY"
+
+restart_ccm_knox_tunnel_service:
+  cmd.run:
+    - name: systemctl restart ccm-tunnel@KNOX
+    - onlyif: systemctl list-unit-files | grep "^ccm-tunnel@KNOX"
```

Comment:
I guess it could be moved to the common salt package instead of it being only part of the freeipa salt state, right? I mean these will be needed for dl/dh proxy modification.